src/main/java/uk/ac/cam/cl/dtg/segue/api/userAlerts/UserAlertsWebSocket.java

```diff
@@ -43,12 +55,34 @@
     private Session session;
     private static ObjectMapper objectMapper = new ObjectMapper();
 
-    public static ConcurrentHashMap<Long, ConcurrentLinkedQueue<UserAlertsWebSocket>> connectedSockets = new ConcurrentHashMap<>();
-    private static Long websocketsOpened = 0L;
-    private static Long websocketsClosed = 0L;
+    private static final int MAX_CONCURRENT_WEB_SOCKETS_PER_USER = 10;
+
+    // Named unsafeConnectedSockets because, although non-aggregate operations on the concurrent hash map are fine,
+    // operations on the user sets of websockets are unsafe unless used with the matching user lock.
+    private static Map<Long, Set<UserAlertsWebSocket>> unsafeConnectedSockets = Maps.newConcurrentMap();
+    private static final int MAX_NUMBER_OF_CONCURRENT_USER_TAB_OPERATIONS = 200; // ~ max number of concurrent users
```

Comment:
Huh? If this comment is accurate, this seems a strange place to put a limit on the number of concurrent users!
Is this "concurrently opening and closing WebSockets at any particular instant", or "concurrently have WebSockets open"?

Reply:
It is the "concurrently opening, closing and sending user messages down a WebSocket at any particular instant".
If we can find a more explanatory name, we should definitely use it.

Reply:
This probably should be a config value in the settings file?

Reply:
Moved MAX_CONCURRENT_WEB_SOCKETS_PER_USER  to be a configuration setting. The other, MAX_NUMBER_OF_CONCURRENT_USER_TAB_OPERATIONS , is used to decide the value of a static property of the class so does not play quite as nicely with our guice injected property reading processes.

The way around it would be to have a singleton instance of a UserAlertsManager class which holds the static stuff and instances of UserAlerts.